### ISSUE TITLE
Backpropagation

### ISSUE THREAD

**Issue by Wheelspawn** [Wheelspawn/Python-Neural-Networks#1]
- Backpropagation algorithm must be correct and generalize to an arbitrary amount of layers.
- Inertia term must be added (to prevent gradient descent from adjusting the weights too quickly or slowly).
- User should be allowed to choose between stochastic learning (each weight is updated as soon as its delta is calculated) and batch learning (where all the deltas are added to a list and then iterated through the network).
